### DISCUSSION TITLE
Is there an option to restrict the extensions entered by the user?

### DISCUSSION THREAD

**Discussion by iMuFeng** [outline/rich-markdown-editor#229] [General]
For example, only allow users to use `Text`,  `Paragraph` extensions.

```js
return new ExtensionManager(
  [
    new Doc(),
    new Text(),
    new Paragraph()
  ]
)
```

**Comment by pradel** [comment]
Would love to have this option too, I would like to limit the extensions a user can use, maybe we the `extensions` prop could be changed to `Extensions[] | (coreExtensions) => Extensions[]` that way a user could provide a function to filter the core extensions they don't use. I would be up to submit a pr if you think that agree with this change @tommoor.

**Reply by tommoor** [reply]
I'm open to it, there is another PR open to disable menu items (https://github.com/outline/rich-markdown-editor/pull/312) but I think a more generic approach that allows disabling of extensions and then have the menus reflect that is the way forward.

A filter approach could work, and I like the backward compatible suggestion. The main problem you'll find is that there are some assumptions built into the menus that the core extensions are always there. So the menus will need updating to only show functionality when the associated extension is available.
